python/lsst/resources/_resourcePath.py

```diff
@@ -426,6 +430,10 @@ def split(self) -> tuple[ResourcePath, str]:
         Equivalent to `os.path.split` where head preserves the URI
         components.
         """
+        if self.isdir():
+            # This is a directory so must return itself and the empty string.
+            return self, ""
+
```

Comment:
For `a/b` it will return `a/b` for the directory if both `a` and `b` happen to exist relative to cwd and `b` is a directory. This does mean that you can get a different answer based on what directory you happen to be sitting in for schemeless. Changing semantics such that people are required to provide the trailing slash for a directory is going to break things a little as described above with zsh dropping `/` on autocomplete on the command line.

os.path and pathlib assume things based on a trailing slash that they sometimes drop in other scenarios.

Reply:
Offline we have agreed to leave `split()` calling `isdir()` and to deprecate `split`, `dirname`, and `basename` on another ticket. `pathlib.Path` has a consistent approach of `parent` and `name` and we will switch to that and if someone wants to know if it is a directory they call `isdir` themselves.